transcom/mymove
# [MB-10957] Add new columns for duty station rename

## [Jira ticket](https://dp3.atlassian.net/browse/MB-10957) for this change

## Summary
We want to rename:
- `service_members.duty_station_id` to `duty_location_id`
- `orders.origin_duty_station_id` to `origin_duty_location_id`
- `orders.new_duty_station_id` to `new_duty_location_id`

In this PR, I will be adding columns with the new names, so that we can gradually switch over to using them, and then drop the old versions of the columns without downtime.
I'm also adding triggers to keep the new columns in sync with the old ones.

my plan is
- Migration 1: create columns with new names and triggers to fill them in *<- You are here*
- Migration 2: copy all values from old columns to new columns
- Reference the new columns from the application code
- Migration 3: drop old columns and triggers.

I'd love to know if there's a way to cut down the number of steps here.

### Is there anything you would like reviewers to give additional scrutiny?

I don't think we use any triggers anywhere yet. Do we need to grant additional permissions?
Should I test this in exp?

## Setup to Run Your Code

Run migrations and populate devseed data
```sh
make db_dev_reset
make db_dev_e2e_populate
```
Make sure that the columns have been kept in sync:
```sql
-- Query for any records with mismatched ids (these should give no results):
select * from orders where origin_duty_station_id <> origin_duty_location_id or orders.new_duty_station_id <> orders.new_duty_location_id;
```
```sql
select * from service_members where duty_station_id <> duty_location_id;
```
If those queries returned no results (as expected) you can run the following queries just to do a quick spot check to make sure things look OK. This is to catch things like the DB being empty somehow or the test queries being wrong.
```sql
select id, origin_duty_station_id, origin_duty_location_id, new_duty_station_id, new_duty_location_id
from orders;

select id, duty_station_id, duty_location_id
from service_members;
```

## Manually test
```
make server_run
make client_run
```
Create a customer, then update their origin and new duty locations as a TOO or services counselor.
Make sure that the changes saved, and then check the DB to make sure `service_members.duty_location_id` `service_members.duty_station_id`, `orders.origin_duty_station_id`, and `orders.origin_duty_location_id` are all equal and that `orders.new_duty_station_id = orders.new_duty_location_id`

```
select m.id,  m.locator, sm.duty_station_id, sm.duty_location_id, o.origin_duty_station_id, o.origin_duty_location_id, o.new_duty_station_id, o.new_duty_location_id
from moves m
         join orders o on m.orders_id = o.id
         join service_members sm ON o.service_member_id = sm.id
where m.locator = 'YOUR MOVE LOCATOR';

```

## Verification Steps for Author

These are to be checked by the author.


## Verification Steps for Reviewers

### Database

#### Any new migrations/schema changes:
